gammapy/modeling/parameter.py

```diff
@@ -755,3 +799,85 @@ def __exit__(self, type, value, traceback):
             if self.restore_values:
                 par.value = value
             par.frozen = frozen
+
+
+class PriorParameter(Parameter):
+    def __init__(
+        self,
+        name,
+        value,
+        unit="",
+        scale=1,
+        min=np.nan,
+        max=np.nan,
+        error=0,
+    ):
+        if not isinstance(name, str):
+            raise TypeError(f"Name must be string, got '{type(name)}' instead")
+
+        self._name = name
+        self.scale = scale
+        self.min = min
+        self.max = max
+        self._error = error
+        if isinstance(value, u.Quantity) or isinstance(value, str):
+            val = u.Quantity(value)
+            self.value = val.value
+            self.unit = val.unit
+        else:
+            self.factor = value
+            self.unit = unit
+        self._type = "prior"
+
+    def to_dict(self):
+        """Convert to dict."""
+        output = {
+            "name": self.name,
+            "value": self.value,
+            "unit": self.unit.to_string("fits"),
+            "error": self.error,
+            "min": self.min,
+            "max": self.max,
+        }
+        return output
+
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}(name={self.name!r}, value={self.value!r}, "
+            f"factor={self.factor!r}, scale={self.scale!r}, unit={self.unit!r}, "
+            f"min={self.min!r}, max={self.max!r})"
+        )
+
+
+class PriorParameters(Parameters):
```

Comment:
It is not totally clear to me whether this class is really needed, as one does not need to interact with the list of all prior parameters set. Yet I suppose the `from_dict` method to build the `Prior` justifies it.